*: add storage class document and aliases (#1484)
pingcap/docs: dev/tidb-in-kubernetes/reference/configuration/storage-class.md

For more information, refer to [Kubernetes local storage](https://kubernetes.io/docs/concepts/storage/volumes/#local) and [local-static-provisioner document](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner#overview).

### Best Practices

- The path of local PV is the unique identifier of local storage on the node. To avoid conflict, it is recommended to use the UUID of the device to generate a unique path.
- For IO isolation, a whole physical disk per volume is recommended.
- For capacity isolation, a separate partition per volume is recommended.

Refer to [Best Practices](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/docs/best-practices.md) for more information with local PV in Kubernetes.

## Data Security

In general, after the PVC is used up and deleted, the PV bound to it is reclaimed and placed in the resource pool for scheduling by the provisioner. To avoid accidental data loss, you can globally configure `StorageClass`'s reclaim policy to `Retain` or only change a single PV's reclaim policy to `Retain`. In `Retain` mode, PV is not automatically reclaimed.

> - Global configuration

Reviewer: ```suggestion
- Configure globally
```

>     The `StorageClass` reclaim policy cannot be modified once it is created, so it can only be set at creation time. If it is not set when created, you can create another `StorageClass` of the same provisioner. For example, the default reclaim policy of `StorageClass` on Google Kubernetes Engine (GKE) is `Delete`. You can create another `StorageClass` named `pd-standard` with reclaim policy as `Retain`, and change the `storageClassName` of the corresponding component to `pd-standard` when creating TiDB clusters.

Reviewer: ```suggestion
    The `StorageClass`'s reclaim policy cannot be modified once it is created, so it can only be set at creation time. If it is not set when created, you can create another `StorageClass` of the same provisioner. For example, the default reclaim policy of `StorageClass` on Google Kubernetes Engine (GKE) is `Delete`. You can create another `StorageClass` named `pd-standard` with its reclaim policy as `Retain`, and change the `storageClassName` of the corresponding component to `pd-standard` when creating a TiDB cluster.
```

    {{< copyable "" >}}

    ```yaml
    apiVersion: storage.k8s.io/v1
    kind: StorageClass
    metadata:
      name: pd-standard
    parameters:
       type: pd-standard
    provisioner: kubernetes.io/gce-pd
    reclaimPolicy: Retain
    volumeBindingMode: Immediate
    ```

- Configure a single PV

    {{< copyable "shell-regular" >}}

    ```shell
    kubectl patch pv <pv-name> -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
    ```

> **Note:**
>
> By default, TiDB Operator automatically changes the PV recliam policy of PD and TiKV to `Retain` to ensure data security.

{{< copyable "shell-regular" >}}

```shell
kubectl patch pv <pv-name> -p '{"spec":{"persistentVolumeReclaimPolicy":"Delete"}}'
```

> 

Reviewer: ```suggestion

(missing sentence)

```
Seems that one sentence is missing here.

---

# Persistent Storage Class Configuration in Kubernetes

TiDB cluster components such as PD, TiKV, TiDB monitoring, TiDB Binlog and `tidb-backup` require the persistent storage of data. To persist the data in Kubernetes, you need to use [PersistentVolume (PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/). Kubernetes supports several types of [storage classes](https://kubernetes.io/docs/concepts/storage/volumes/), which are mainly divided into two parts:

- Network storage

    The network storage medium is not on the current node, but is mounted to the node through the network. Generally, there are redundant replicas to guarantee high availability. When the node fails, the corresponding network storage can be re-mounted to another node for further use.

- Local storage

    The local storage medium is on the current node, and typically can provide lower latency than the network storage. Because there are no redundant replicas, once the node fails, data might be lost. If it is an IDC server, data can be restored to a certain extent. If it is a virtual machine using the local disk on the public cloud, data **cannot** be retrieved after the node fails.

> PVs are created automatically by the system administrator or volume provisioner. PVs and Pods are bound by[PersistentVolumeClaim (PVC)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims). Users request for using a PV through a PVC instead of creating a PV directly. The corresponding volume provisioner creates a PV that meets the requirements of PVC and then binds the PV to the PVC.

Reviewer: ```suggestion
PVs are created automatically by the system administrator or volume provisioner. PVs and Pods are bound by [PersistentVolumeClaim (PVC)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims). Users request for using a PV through a PVC instead of creating a PV directly. The corresponding volume provisioner creates a PV that meets the requirements of PVC and then binds the PV to the PVC.
```

2. Install the `local-volume-provisioner` program. See the [Helm installation procedure](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/tree/master/helm) for reference.

For more information, refer to [Kubernetes local storage](https://kubernetes.io/docs/concepts/storage/volumes/#local) and [local-static-provisioner document](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner#overview).

### Best practices

- The path of a local PV is the unique identifier for the local volume. To avoid conflicts, it is recommended to use the UUID of the device to generate a unique path.
- For I/O isolation, a dedicated physical disk per volume is recommended to ensure hardware-based isolation.
- For capacity isolation, a dedicated partition per volume is recommended.

Refer to [Best Practices](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/docs/best-practices.md) for more information on local PV in Kubernetes.

> ## Data security

Reviewer 2: security -> safety

- Configure globally:

    The `StorageClass`'s reclaim policy cannot be modified once it is created, so it can only be set at creation time. If it is not set when created, you can create another `StorageClass` of the same provisioner. For example, the default reclaim policy of `StorageClass` on Google Kubernetes Engine (GKE) is `Delete`. You can create another `StorageClass` named `pd-standard` with its reclaim policy as `Retain`, and change the `storageClassName` of the corresponding component to `pd-standard` when creating a TiDB cluster.

    {{< copyable "" >}}

    ```yaml
    apiVersion: storage.k8s.io/v1
    kind: StorageClass
    metadata:
      name: pd-standard
    parameters:
       type: pd-standard
    provisioner: kubernetes.io/gce-pd
    reclaimPolicy: Retain
    volumeBindingMode: Immediate
    ```

- Configure a single PV:

    {{< copyable "shell-regular" >}}

    ```shell
    kubectl patch pv <pv-name> -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
    ```

> **Note:**
>

> > By default, TiDB Operator automatically changes the PV reclaim policy of PD and TiKV to `Retain` to ensure data security.

Reviewer 2: security -> safety
